### PR TITLE
S 1

### PR DESCRIPTION
**Pull request by varuntaak**

### CODE REVIEW

File: src/app/schedule_structure.py [hunk 1]
```diff
@@ -54,15 +56,27 @@ def set(self, activity, hour, value):
...
 
         '''
         Value incrementer, increment the value by 1 at the postion.
         :param activity_type: Type of the activity
         :param hour: An hour between 1 to 24.
         '''
 
-        c = self.get(activity, hour)
-        self.set(activity, hour, c + 1)
+        c = self._get(activity, hour)
```
Comment (sapkotashardul) [inline src/app/schedule_structure.py]:
```suggestion
        cell_value = self._get(activity, hour) 
```

File: src/assistant/Assistant.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+from typing import List
+
+import math
+
+from app.schedule_structure import ScheduleStructure
+from model.day import Day
+from value.acceptance import Acceptance
+
+
+class Assistant():
```
Comment (sapkotashardul) [inline src/assistant/Assistant.py]: I think before merging this class, we could add some documentation to it? For each function the param type, description and same for the return values?

File: src/assistant/Assistant.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    def __init__(self, sch: ScheduleStructure, days: List[Day]):
+        self.sch = sch
+        self.days = days
+        self.acceptance = Acceptance(schedule=sch)
+
+    def get_best_slots(self, activity: Enum):
+        fr = {}
+        p_list = []
```
Comment (sapkotashardul) [inline src/assistant/Assistant.py]:
```suggestion
        probability_list = []
```

File: src/assistant/Assistant.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+class Assistant():
+
+    def __init__(self, sch: ScheduleStructure, days: List[Day]):
+        self.sch = sch
+        self.days = days
+        self.acceptance = Acceptance(schedule=sch)
+
+    def get_best_slots(self, activity: Enum):
+        fr = {}
```
Comment (sapkotashardul) [inline src/assistant/Assistant.py]: define fr as free_slots?

File: src/assistant/Assistant.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        p_list = []
+        for di, d in enumerate(self.days):
```
Comment (sapkotashardul) [inline src/assistant/Assistant.py]:
```suggestion
        for day_index, day in enumerate(self.days):
```

**Comment by varuntaak** [comment]
@sapkotashardul Thanks for the comments!
I think for local variables we practice to keep it short else the code will look too verbose. Interfaces and Global variables may go to a wordy description. For example, "di", it has no purpose other than helping control to set the value to the final result. So knowing that it is "day_index" makes a little sense when I already know that I am iterating on days. On the other hand, the "s" could be named more descriptively as "slots" to give more sense on what are we iterating on further in the loop. It is also a practice of clean code. But may comes with practice to decide what variable names should be wordy and what are not. 
Having a strategy to name every variable descriptively will not work too smoothly either. The code will look verbose with lots of information in it, which is not necessary from the algorithmic point of view. 
For now, I am pushing the missing documentation. We will see these renaming of the variables in more like refactoring if needed. Let me know your view.

**Comment by varuntaak** [comment]
@sapkotashardul Thanks for the comments!
I think for local variables we practice to keep it short else the code will look too verbose. Interfaces and Global variables may go to a wordy description. For example, "di", it has no purpose other than helping control to set the value to the final result. So knowing that it is "day_index" makes a little sense when I already know that I am iterating on days. On the other hand, the "s" could be named more descriptively as "slots" to give more sense on what are we iterating on further in the loop. It is also a practice of clean code. But may comes with practice to decide what variable names should be wordy and what are not. 
Having a strategy to name every variable descriptively will not work too smoothly either. The code will look verbose with lots of information in it, which is not necessary from the algorithmic point of view. 
For now, I am pushing the missing documentation. We will see these renaming of the variables in more like refactoring if needed. Let me know your view.

File: src/app/schedule_structure.py (hunk 1 above)
Comment (varuntaak) [reply src/app/schedule_structure.py]: @sapkotashardul Thanks for the comments!
I think for local variables we practice to keep it short else the code will look too verbose. Interfaces and Global variables may go to a wordy description. For example, "di", it has no purpose other than helping control to set the value to the final result. So knowing that it is "day_index" makes a little sense when I already know that I am iterating on days. On the other hand, the "s" could be named more descriptively as "slots" to give more sense on what are we iterating on further in the loop. It is also a practice of clean code. But may comes with practice to decide what variable names should be wordy and what are not. 
Having a strategy to name every variable descriptively will not work too smoothly either. The code will look verbose with lots of information in it, which is not necessary from the algorithmic point of view. 
For now, I am pushing the missing documentation. We will see these renaming of the variables in more like refactoring if needed. Let me know your view.

File: src/assistant/Assistant.py (hunk 2 above)
Comment (varuntaak) [reply src/assistant/Assistant.py]: added

**Comment by varuntaak** [comment]
@sapkotashardul Thanks for the comments!
I think for local variables we practice to keep it short else the code will look too verbose. Interfaces and Global variables may go to a wordy description. For example, "di", it has no purpose other than helping control to set the value to the final result. So knowing that it is "day_index" makes a little sense when I already know that I am iterating on days. On the other hand, the "s" could be named more descriptively as "slots" to give more sense on what are we iterating on further in the loop. It is also a practice of clean code. But may comes with practice to decide what variable names should be wordy and what are not. 
Having a strategy to name every variable descriptively will not work too smoothly either. The code will look verbose with lots of information in it, which is not necessary from the algorithmic point of view. 
For now, I am pushing the missing documentation. We will see these renaming of the variables in more like refactoring if needed. Let me know your view.

**Comment by varuntaak** [comment]
@sapkotashardul Thanks for the comments!
I think for local variables we practice to keep it short else the code will look too verbose. Interfaces and Global variables may go to a wordy description. For example, "di", it has no purpose other than helping control to set the value to the final result. So knowing that it is "day_index" makes a little sense when I already know that I am iterating on days. On the other hand, the "s" could be named more descriptively as "slots" to give more sense on what are we iterating on further in the loop. It is also a practice of clean code. But may comes with practice to decide what variable names should be wordy and what are not. 
Having a strategy to name every variable descriptively will not work too smoothly either. The code will look verbose with lots of information in it, which is not necessary from the algorithmic point of view. One approach is we can refactor the code for these name related issues later when the algorithms are final and stable. Let me know your view?

**Comment by varuntaak** [comment]
@sapkotashardul Thanks for the comments!
I think for local variables we practice to keep it short else the code will look too verbose. Interfaces and Global variables may go to a wordy description. For example, "di", it has no purpose other than helping control to set the value to the final result. So knowing that it is "day_index" makes a little sense when I already know that I am iterating on days. On the other hand, the "s" could be named more descriptively as "slots" to give more sense on what are we iterating on further in the loop. It is also a practice of clean code. But may comes with practice to decide what variable names should be wordy and what are not. 
Having a strategy to name every variable descriptively will not work too smoothly either. The code will look verbose with lots of information in it, which is not necessary from the algorithmic point of view. One approach is we can refactor the code for these name related issues later when the algorithms are final and stable. Let me know your view?

**Comment by varuntaak** [comment]
got some strange merge conflicts here, closing it down for now. I will raise another request.

**Comment by sapkotashardul** [comment]
Yeah, I think with that context what you said definitely makes sense! But should we make sure that the context is always clear to figure out what the short variable names mean?

**Pull request closed without merging**